openmdao/approximation_schemes/approximation_scheme.py

```diff
@@ -296,6 +301,13 @@ def _init_approximations(self, system):
     def _compute_approximations(self, system, jac, total, under_cs):
         from openmdao.core.component import Component
 
+        if hasattr(self, "progress_out_stream"):
```

Comment:
You don't need this entire if block, if the value of _progress_out is being set in the approximation scheme instead of the dictionary it lives in, as per the other changes I marked.